Rx/v2/test/operators/merge_delay_error.cpp

```diff
@@ -0,0 +1,390 @@
+                REQUIRE(required == actual);
+            }
+
+            THEN("there was one subscription and one unsubscription to the ys2"){
+                auto required = rxu::to_vector({
+                    on.subscribe(400, 450)
+                });
+                auto actual = ys2.subscriptions();
+                REQUIRE(required == actual);
+            }
+
+            THEN("there was one subscription and one unsubscription to the ys3"){
+                auto required = rxu::to_vector({
+                    on.subscribe(500, 650)
+                });
+                auto actual = ys3.subscriptions();
+                REQUIRE(required == actual);
+            }
+        }
+    }
+}
+
+SCENARIO("variadic merge completes with error", "[merge][join][operators]"){
+    GIVEN("1 hot observable with 3 cold observables of ints."){
+        auto sc = rxsc::make_test();
+        auto w = sc.create_worker();
+        const rxsc::test::messages<int> on;
+        const rxsc::test::messages<rx::observable<int>> o_on;
+
+        auto ys1 = sc.make_cold_observable({
+            on.next(10, 101),
+            on.next(20, 102),
+            on.next(110, 103),
+            on.next(120, 104),
+            on.next(210, 105),
+            on.next(230, 107),
+            on.completed(240)
+        });
+
+        auto ys2 = sc.make_cold_observable({
+            on.next(10, 201),
+            on.next(20, 202),
+            on.next(30, 203),
+            on.error(40, std::runtime_error("merge_delay_error on_error from ys2")),
+            on.next(50, 205),
+            on.completed(60)
+        });
+
+        auto ys3 = sc.make_cold_observable({
+            on.next(10, 301),
+            on.next(20, 302),
+            on.next(30, 303),
+            on.next(40, 304),
+            on.next(120, 305),
+            on.completed(150)
+        });
+
+        WHEN("each int is merged"){
+
+            auto res = w.start(
+                [&]() {
+                    return ys1
+                        .merge_delay_error(ys2, ys3);
+                }
+            );
+
+            rx::composite_exception ex;
+            THEN("the output contains merged ints"){
+                auto required = rxu::to_vector({
+                    on.next(210, 101),
+                    on.next(210, 201),
+                    on.next(210, 301),
+                    on.next(220, 102),
+                    on.next(220, 202),
+                    on.next(220, 302),
+                    on.next(230, 203),
+                    on.next(230, 303),
+                    on.next(240, 304),
+                    on.next(310, 103),
+                    on.next(320, 104),
+                    on.next(320, 305),
+                    on.next(410, 105),
+                    on.next(430, 107),
+                    on.error(440, ex)
+                });
+                auto actual = res.get_observer().messages();
+                REQUIRE(required == actual);
+            }
```

Comment:
it would be nice to be able to check that the exception list had the correct exceptions stored.. but I think that would require a lot of work in the test infrastructure. something for the future.

Reply:
yes, I agree - I also spotted that the test does not cover the actual exception assertation. I'll take a look at that later